src/swoop/api/routers/processes.py

```diff
@@ -109,46 +95,83 @@ async def get_process_description(
     """
     retrieve a process description
     """
-    workflows = (
-        request.app.state.workflows.get("workflows")
-        if request.app.state.workflows.get("workflows")
-        else []
-    )
+    workflows = request.app.state.workflows
 
-    if process_id and len(workflows) > 0:
-        workflows = list(
-            filter(
-                lambda x: process_id == x["name"],
-                request.app.state.workflows["workflows"],
-            )
-        )
-    if not workflows:
+    try:
+        workflow = workflows[process_id]
+    except KeyError:
         raise HTTPException(status_code=404, detail="Process not found")
 
-    return to_process_summary(workflows[0])
-
-
-@router.get("/{process_id}/definition")
-async def get_process_definition(process_id: str = Path(..., alias="processID")) -> str:
-    """
-    retrieve a process definition
-    """
-    pass
+    return to_process_summary(workflow)
 
 
 @router.post(
     "/{process_id}/execution",
-    response_model=InlineResponse200,
+    response_model=StatusInfo,
     responses={
         "201": {"model": StatusInfo},
         "404": {"model": APIException},
-        "500": {"model": APIException},
+        "422": {"model": APIException},
     },
+    status_code=201,
 )
-def execute_process(
-    process_id: str = Path(..., alias="processID"), body: Execute = ...
-) -> InlineResponse200 | StatusInfo | APIException:
+async def execute_process(
+    process_id: str, request: Request, body: Execute
+) -> StatusInfo | APIException:
     """
     execute a process.
     """
-    pass
+
+    wf_name = body.dict().get("inputs").get("payload").get("process")[0].get("workflow")
```

Comment:
This line assumes `process` in the payload is an array, but the model supports `process` being an array or a map.

After further discussion with @matthewhanson, we reached the conclusion that we will no longer support the map, and `process` should always be an array. TBD is whether this should now be renamed to `processes` or something, but for now I think just change the model so `process` is always an array and this is fine.

Reply:
Fixed